cmake/faceliftMacros.cmake

```diff
@@ -403,57 +414,130 @@ macro(_facelift_add_target_start additionalOptions additionalOneValueArgs additi
+        string(REPLACE "-" "_" LIB_PREPROCESSOR_DEFINITION "${INTERFACE_TARGET_NAME}_LIBRARY")
+        target_compile_definitions(${IMPLEMENTATION_TARGET_NAME} PRIVATE ${LIB_PREPROCESSOR_DEFINITION} ${ARGUMENT_PRIVATE_DEFINITIONS})
+
+        target_include_directories(${IMPLEMENTATION_TARGET_NAME} PRIVATE ${ARGUMENT_PRIVATE_INCLUDE_DIRECTORIES})
+
+    endif()
 
+    unset(INTERFACE_IF_INTERFACE_TARGET)
+    get_target_property(target_type ${INTERFACE_TARGET_NAME} TYPE)
+    if(target_type STREQUAL "INTERFACE_LIBRARY")
+        set(INTERFACE_TARGET_MODIFIER INTERFACE)
+        set(INTERFACE_IF_INTERFACE_TARGET INTERFACE)
+    else()
+        set(INTERFACE_TARGET_MODIFIER PUBLIC)
+    endif()
+
+    target_compile_definitions(${INTERFACE_TARGET_NAME} ${INTERFACE_TARGET_MODIFIER} ${ARGUMENT_PUBLIC_DEFINITIONS})
+
+    if(ARGUMENT_PROPERTIES)
+        set_target_properties(${IMPLEMENTATION_TARGET_NAME} PROPERTIES ${ARGUMENT_PROPERTIES})
     endif()
 
     # We assume every lib links against QtCore at least
-    target_link_libraries(${TARGET_NAME} ${__INTERFACE} Qt5::Core ${ARGUMENT_LINK_LIBRARIES})
+    target_link_libraries(${INTERFACE_TARGET_NAME} ${INTERFACE_IF_INTERFACE_TARGET} Qt5::Core ${ARGUMENT_LINK_LIBRARIES})
 
 endmacro()
 
+
+
 function(facelift_add_library TARGET_NAME)
 
-    _facelift_add_target_start("STATIC;SHARED;OBJECT" "" "HEADERS_NO_INSTALL;HEADERS_GLOB_NO_INSTALL;HEADERS_GLOB_RECURSE_NO_INSTALL;PUBLIC_HEADER_BASE_PATH" ${ARGN})
+    _facelift_parse_target_arguments("STATIC;SHARED;OBJECT;MODULE;MONOLITHIC_SUPPORTED" "" "HEADERS_NO_INSTALL;HEADERS_GLOB_NO_INSTALL;HEADERS_GLOB_RECURSE_NO_INSTALL;PUBLIC_HEADER_BASE_PATH;MONOLITHIC_LINK_LIBRARIES" ${ARGN})
 
     unset(__INTERFACE)
     if(ARGUMENT_INTERFACE)
+        _facelift_add_target_start(${TARGET_NAME})
         add_library(${TARGET_NAME} INTERFACE)
         set(__INTERFACE INTERFACE)
+        _facelift_add_target_finish(${TARGET_NAME} ${TARGET_NAME})
     else()
-        if(NOT ALL_SOURCES)
-            # We create an INTERFACE library if the library contains no source file
-            set(__INTERFACE INTERFACE)
-            set(LIBRARY_TYPE INTERFACE)
-        endif()
         if(${ARGUMENT_STATIC})
             set(LIBRARY_TYPE STATIC)
         endif()
         if(${ARGUMENT_OBJECT})
             set(LIBRARY_TYPE OBJECT)
         endif()
+        if(${ARGUMENT_MODULE})
+            set(LIBRARY_TYPE MODULE)
+        endif()
         if(${ARGUMENT_SHARED})
             set(LIBRARY_TYPE SHARED)
         endif()
-        add_library(${TARGET_NAME} ${LIBRARY_TYPE} ${ALL_SOURCES})
-        if(ENABLE_LTO)
-           if(NOT "${LIBRARY_TYPE}" STREQUAL "INTERFACE")
-               set_property(TARGET ${TARGET_NAME} PROPERTY INTERPROCEDURAL_OPTIMIZATION TRUE)
-           endif()
+
+        if(MONOLITHIC_BUILD AND ${ARGUMENT_MONOLITHIC_SUPPORTED})
+
+#            message(WARNING "Enabling MONOLITHIC for lib : ${TARGET_NAME}")
```

Comment:
Please remove commented out line or add reason why it is commented out.
Personally I would keep info message here.

Reply:
Removed